lib/rbac/filterer.rb

```diff
@@ -337,23 +337,29 @@ def search(options = {})
       #
       # NOTE: if supported_by_sql is false then apply_limit_in_sql is also false
       if search_filter && targets && (!exp_attrs || !exp_attrs[:supported_by_sql])
-        rejects     = targets.reject { |obj| matches_search_filters?(obj, search_filter, tz) }
-        auth_count -= rejects.length unless options[:skip_counts]
-        targets -= rejects
+        targets = targets.lazy.select { |obj| matches_search_filters?(obj, search_filter, tz) }
```

Comment:
note. this was changed from a negative to a positive.

```diff
- rejects = targets.reject { miq_expression }
- targets = targets - rejects
+ targets = targets.select { miq_expression}

- auth_count -= rejects.length
+ auth_count = targets.length
```

Can't remember why this was done this way.
Looking at the counts seems like it should be the same way